### PR TITLE
[java] Ensure retry mechanism does not swallow an exception

### PR DESCRIPTION
**Pull request by pujagani**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Ensure retry mechanism does not swallow an exception

### Motivation and Context

Related to https://github.com/SeleniumHQ/selenium/issues/12558#issuecomment-1738871816.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [X] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: java/src/org/openqa/selenium/remote/http/RetryRequest.java [hunk 1]
```diff
@@ -30,16 +30,62 @@
...
 public class RetryRequest implements Filter {
 
   private static final Logger LOG = Logger.getLogger(RetryRequest.class.getName());
-  private static final AtomicReference<HttpResponse> fallBackResponse = new AtomicReference<>();
 
-  private static final Fallback<Object> fallback = Fallback.of(fallBackResponse::get);
+  private static final Fallback<Object> fallback =
+      Fallback.of(
+          executionAttemptedEvent -> {
+            if (executionAttemptedEvent.getLastException() != null) {
```
Comment (shs96c) [inline java/src/org/openqa/selenium/remote/http/RetryRequest.java]: Given the length of this, I'd suggest pulling out it out into a `private` method in this class. It'll make it easier to visually figure out the fields of the class from the methods it contains.

**Review by shs96c — APPROVED** [review]
One comment, but looks great. Love the tests.

File: java/src/org/openqa/selenium/remote/http/RetryRequest.java (hunk 1 above)
Comment (pujagani) [reply java/src/org/openqa/selenium/remote/http/RetryRequest.java]: Ahh makes sense. Thank you! Will do that.

**Comment by titusfortner** [comment]
Does this fix #12558 or just related to the discussion there?

**Review by diemol — APPROVED** [review]

**Comment by diemol** [comment]
https://github.com/SeleniumHQ/selenium/issues/12558 was fixed via https://github.com/SeleniumHQ/selenium/commit/5f3f7d096a75700e055effbe13cbc22117014e2c, but the issue diverted to another error which is getting fixed with this PR.

**Comment by joerg1985** [comment]
@pujagani thanks for fixing this 😀

**Pull request merged**